internal/logging/logging.go

```diff
@@ -237,6 +237,26 @@ func ReadTail() string {
 	return logTail.Read()
 }
 
+var svcTailProvider func() string
+
+// SetSvcTailProvider registers the given function as providing the state-svc log tail.
+// Fetching that log via pkg/platform/model is not possible here due to an import cycle, so the
+// given function is expected to do that.
+func SetSvcTailProvider(f func() string) {
+	svcTailProvider = f
+}
+
+// ReadSvcTail returns the tail end of the state-svc log, just like ReadTail().
+// This exists in order to have Rollbar send both state logs and state-svc logs together.
+// If no provider function was given earlier, this function returns the empty string, so calling it
+// from state-svc is safe. (i.e. state-svc is sending its own error log to Rollbar.)
+func ReadSvcTail() string {
+	if svcTailProvider != nil {
+		return svcTailProvider()
```

Comment:
I suggest that the truncation should be moved here in the same way that the ring buffer does its work in this package.

Reply:
I don't follow. The state tool supplies a svcTailProvider that asks the service to send its truncated log tail (which is the contents of the service's own ring buffer). Are you asking the service to send its entire log over the wire and have the truncation happen here in the state tool process?

Reply:
Ah, yeah, that sounds rough. In that case I'd send the truncation length as an argument, but I'll leave it to you to decide if that's worthwhile. Favoring the truncation being done before being transferred is surely preferable, imo.

Reply:
Right, and it's also a constant value, just like in the state tool: https://github.com/ActiveState/cli/blob/f569d74340062440556372e7a539a3a33a5e87ff/internal/logging/logging.go#L219

Reply:
I see what's going on... I think I just don't like the way that the truncation is being communicated in the rollbar package. No worries: If it's worked fine so far, let's leave it. IOW, ignore this thread.